### PR TITLE
Correct svn mission's diff step to remove 500 error [Issue #1399]

### PR DESCRIPTION
**Pull request by willingc**
Removed a try block and un-pep8ed view file for svn mission. The diff step of the mission now redirected properly on my local system. It is not throwing a 500 error locally.

### CODE REVIEW

**Comment by willingc** [comment]
Travis CI is broken :( https://twitter.com/traviscistatus/status/592902357144498176

Hoping this PR tests cleanly after Travis rolls back their upgrade.

**Comment by willingc** [comment]
Closes #1399 if Travis blesses.

**Comment by willingc** [comment]
@paulproteus Based on our IRC discussion, I have not removed the try block around 'form.commit_diff()' in 'def diff_submit(request):'. Instead, I refactored what the try block contains. I've also added a new test and added some docstrings. 

This PR is ready for review. Let me know if you want me to change anything. Tested locally as well and no 500 error :)

File: mysite/missions/svn/views.py [hunk 1]
```diff
@@ -178,4 +186,5 @@ class Commit(SvnBaseView):
 
 @login_required
 def commit_poll(request):
+    """ Determines if entire mission is completed """
```
Comment (ehashman) [inline mysite/missions/svn/views.py]: +1 new comments and PEP8 compliance!

File: mysite/missions/svn/tests.py [hunk 2]
```diff
@@ -219,6 +219,34 @@ def test_main_page_does_not_complain_about_prereqs(self):
         self.assertTrue(response.context[0]
                         ['mission_step_prerequisites_passed'])
 
+    def test_do_diff_mission_with_invalid_form(self):
```
Comment (ehashman) [inline mysite/missions/svn/tests.py]: +1 new test!!

**Comment by ehashman** [comment]
Travis seems to like this, so I do too. :cake:

**Pull request merged**